html/modules/custom/reliefweb_import/src/Command/ReliefwebImportCommand.php

```diff
@@ -0,0 +1,680 @@
+
+    // Check if the text is wrapped in a CDATA.
+    if (mb_stripos($text, '<![CDATA[') === 0) {
+      $end = mb_strpos($text, ']]>');
+      $text = mb_substr($text, 9, $end !== FALSE ? $end - 9 : NULL);
+    }
+
+    // Check if the content contains some non encoded html tags, in which case
+    // we will assume that the text is non encoded html/markdown. For that we
+    // simply search for a closing tag '</...>'. Otherwise we decode the text.
+    if (preg_match('#</[^>]+>#', $text) !== 1) {
+      $text = html_entity_decode($text, ENT_QUOTES, 'UTF-8');
+    }
+
+    // Convert the text to HTML.
+    /*
+    if (function_exists('check_markup')) {
+    $text = check_markup($text, $format)->__toString();
+    }
+     */
+
+    // We then sanitize the HTML string.
+    $sanitizer = new HtmlSanitizer();
+    $text = $sanitizer->sanitizeHtml($text);
+
+    // Remove embedded content.
+    $text = $this->stripEmbeddedContent($text);
+
+    // Finally we convert it to markdown.
+    $converter = new HtmlConverter();
+    $converter->getConfig()->setOption('strip_tags', TRUE);
+    $converter->getConfig()->setOption('use_autolinks', FALSE);
+    $converter->getConfig()->setOption('header_style', 'atx');
+    $converter->getConfig()->setOption('strip_placeholder_links', TRUE);
+
+    $text = trim($converter->convert($text));
+
+    return $text;
+  }
+
+  /**
+   * Clean a text.
+   *
+   * 1. Replace tabulations with double spaces.
+   * 2. Replace non breaking spaces with normal spaces.
+   * 3. Remove control characters (except line feed).
+   * 4. Optionally, replace line breaks and consecutive whitespaces.
+   * 4. Trim the text.
+   *
+   * @param string $text
+   *   Text to clean.
+   * @param array $options
+   *   Associative array with the following replacement options:
+   *   - line_breaks (boolean): replace line breaks with spaces.
+   *   - consecutive (boolean): replace consecutive whitespaces.
+   *
+   * @return string
+   *   Cleaned text.
+   */
+  protected function cleanText($text, array $options = []) {
+    $patterns = ['/[\t]/u', '/[\xA0]+/u', '/[\x00-\x09\x0B-\x1F\x7F]/u'];
+    $replacements = ['  ', ' ', ''];
+    // Replace (consecutive) line breaks with a single space.
+    if (!empty($options['line_breaks'])) {
+      $patterns[] = '/[\x0A]+/u';
+      $replacements[] = ' ';
+    }
+    // Replace consecutive whitespaces with single space.
+    if (!empty($options['consecutive'])) {
+      $patterns[] = '/\s{2,}/u';
+      $replacements[] = ' ';
+    }
+    return trim(preg_replace($patterns, $replacements, $text));
+  }
+
+  /**
+   * Remove embedded content in html or markdown format from the given text.
+   *
+   * Note: it's using a very basic pattern matching that may not work with
+   * broken html (missing </iframe> ending tag for example)
+   *
+   * @param string $text
+   *   Text to clean.
+   *
+   * @return string
+   *   Cleaned up text.
+   */
+  protected function stripEmbeddedContent($text) {
```

Comment:
This could be moved to the TextHelper: https://github.com/UN-OCHA/rwint9-site/blob/develop/html/modules/custom/reliefweb_utility/src/Helpers/TextHelper.php